pkg/skaffold/kubernetes/context/context.go

```diff
@@ -30,26 +31,54 @@ var (
 )
 
 var (
-	currentConfigOnce sync.Once
-	currentConfig     clientcmdapi.Config
-	currentConfigErr  error
+	kubeConfigOnce sync.Once
+	kubeConfig     clientcmd.ClientConfig
+	kubeContext    string
 )
 
-// ResetCurrentConfig is used by tests
-func ResetCurrentConfig() {
-	currentConfigOnce = sync.Once{}
+// resetConfig is used by tests
+func resetConfig() {
+	kubeConfigOnce = sync.Once{}
 }
 
+// UseKubeContext sets an override for the current context in the k8s config.
+func UseKubeContext(overrideKubeContext string) {
```

Comment:
Same question, wouldn't it be better to have `struct` and assigned methods instead of global vars with simple functions operating those vars?

Reply:
I presume, you want to get rid of global state, did I get that right? I agree that this is always preferable. However, the kubeContext can be overridden at some point, and the sites which need to access, say `CurrentConfig`, don't have access to the correct kubeContext override. That's why this state is currently in a global variable.
Also see https://github.com/GoogleContainerTools/skaffold/pull/2447#discussion_r302246158 for that discussion.